gemrb/core/System/VFS.cpp

```diff
@@ -256,11 +258,95 @@ void PathAppend(path_t& target, const path_t& name)
 	}
 }
 
+static const auto C_LOCALE = std::locale{"C.UTF-8"};
+
+// Returns true if both strings are equal after lower-casing
+static bool strlicmp(const char *a, const char *b) {
+	const unsigned char *itA = reinterpret_cast<const unsigned char*>(a),
+		*itB = reinterpret_cast<const unsigned char*>(b);
+
+	while (*itA != 0) {
+		unsigned char iA = *itA, iB = *itB;
+
+		// a is longer
+		if (iB == 0) {
+			return false;
+		}
+
+		// One is special, the other one isn't
+		if ((iA >= 128 && iB < 128) || (iA < 128 && iB >= 128)) {
+			return false;
+		}
+
+		// ASCII range
+		if (iA < 128) {
+			if (std::tolower(iA) != std::tolower(iB)) {
+				return false;
+			}
+
+			itA++, itB++;
+		} else {
+			// multibyte range (assuming no broken encoding)
+			auto getWidth = [](unsigned char c) -> uint8_t {
```

Comment:
Is this not exactly what `std::mbrlen` does?

Reply:
Seems to depend on `LC_ALL`, so for UTF-8 case I consider this easy enough.

Reply:
See above.